src/pymatgen/core/structure.py

```diff
@@ -1098,9 +1098,9 @@ def __init__(
         self._properties = properties or {}
 
     def __eq__(self, other: object) -> bool:
+        """Define equality by comparing all three attributes: lattice, sites, properties."""
         needed_attrs = ("lattice", "sites", "properties")
 
-        # Return NotImplemented as in https://docs.python.org/3/library/functools.html#functools.total_ordering
```

Comment:
I guess `total_ordering` was applied on `IStructure` at some point but I believe this comment should be removed now:

https://github.com/materialsproject/pymatgen/blob/31f1e1fb8cc1bb517d59ea8e484965bb641c42a0/src/pymatgen/core/structure.py#L1006-L1007

Also current implementation of `IStructure` doesn't fulfill the [prerequisite of applying `total_ordering`](https://docs.python.org/3/library/functools.html#functools.total_ordering):
> The class must define one of __lt__(), __le__(), __gt__(), or __ge__(). In addition, the class should supply an __eq__() method.